lib/mocha.js

```diff
@@ -202,24 +202,24 @@ Mocha.prototype.reporter = function(reporter, reporterOptions) {
         _reporter = require(reporter);
       } catch (err) {
         if (
-          err.code !== 'MODULE_NOT_FOUND' ||
-          err.message.indexOf('Cannot find module') !== -1
+          err.code === 'MODULE_NOT_FOUND' ||
+          err.message.indexOf('Cannot find module') >= 0
         ) {
           // Try to load reporters from a path (absolute or relative)
           try {
-            _reporter = require(path.resolve(process.cwd(), reporter));
+            _reporter = require(path.resolve(reporter));
           } catch (_err) {
-            _err.code !== 'MODULE_NOT_FOUND' ||
-            _err.message.indexOf('Cannot find module') !== -1
-              ? console.warn(sQuote(reporter) + ' reporter not found')
-              : console.warn(
+            _err.code === 'MODULE_NOT_FOUND' ||
+            _err.message.indexOf('Cannot find module') >= 0
+              ? utils.warn(sQuote(reporter) + ' reporter not found')
```

Comment:
Possibly worth a test case for this to prove the logic works, altho i know coverage would flag if not, up to you.

Reply:
I thought I added a test for this.  hmm.

Reply:
https://github.com/mochajs/mocha/pull/4275/files#diff-0a944d7334aa10653542ebf2744ea850R525

and

https://github.com/mochajs/mocha/pull/4275/files#diff-0a944d7334aa10653542ebf2744ea850R525